huaweicloud/services/ecs/resource_huaweicloud_compute_interface_attach.go

```diff
@@ -88,253 +105,351 @@ func ResourceComputeInterfaceAttach() *schema.Resource {
+	}
+	return utils.FlattenResponse(getPortResp)
+}
+
 func resourceComputeInterfaceAttachUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	cfg := meta.(*config.Config)
-	nicClient, err := cfg.NetworkingV2Client(cfg.GetRegion(d))
+	region := cfg.GetRegion(d)
+	vpcClient, err := cfg.NewServiceClient("vpc", region)
 	if err != nil {
-		return diag.Errorf("error creating networking client: %s", err)
+		return diag.Errorf("error creating VPC client: %s", err)
 	}
 
-	var (
-		portId                 = d.Get("port_id").(string)
-		securityGroupIds       = d.Get("security_group_ids").([]interface{})
-		sourceDestCheckEnabled = d.Get("source_dest_check").(bool)
-	)
-	err = updateInterfacePort(nicClient, portId, utils.ExpandToStringList(securityGroupIds), sourceDestCheckEnabled)
+	portID := d.Get("port_id").(string)
+	err = updatePort(vpcClient, portID, d.Get("security_group_ids"), d.Get("source_dest_check").(bool))
 	if err != nil {
-		return diag.Errorf("error updating VPC port (%s): %s", portId, err)
+		return diag.Errorf("error updating ECS NIC port (%s): %s", portID, err)
 	}
 
 	return resourceComputeInterfaceAttachRead(ctx, d, meta)
 }
 
 func resourceComputeInterfaceAttachDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	cfg := meta.(*config.Config)
-	computeClient, err := cfg.ComputeV2Client(cfg.GetRegion(d))
+	region := cfg.GetRegion(d)
+	computeClient, err := cfg.NewServiceClient("ecs", region)
 	if err != nil {
 		return diag.Errorf("error creating compute client: %s", err)
 	}
 
-	instanceId, portId, err := computeInterfaceAttachParseID(d.Id())
+	serverID := d.Get("instance_id").(string)
+
+	deleteNicsHttpUrl := "v1/{project_id}/cloudservers/{server_id}/nics/delete"
+	deleteNicsPath := computeClient.Endpoint + deleteNicsHttpUrl
+	deleteNicsPath = strings.ReplaceAll(deleteNicsPath, "{project_id}", computeClient.ProjectID)
+	deleteNicsPath = strings.ReplaceAll(deleteNicsPath, "{server_id}", serverID)
+
+	nic := make([]map[string]interface{}, 0)
+	nic = append(nic, map[string]interface{}{
+		"id": d.Get("port_id"),
+	})
+	deleteNicOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: map[string]interface{}{
+			"nics": nic,
+		},
+	}
+	// The `DELETE` API use `POST` method actually.
+	_, err = computeClient.Request("POST", deleteNicsPath, &deleteNicOpt)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	stateConf := &resource.StateChangeConf{
-		Pending:    []string{""},
-		Target:     []string{"DETACHED"},
-		Refresh:    computeInterfaceAttachDetachFunc(computeClient, instanceId, portId),
+		Pending:    []string{"DELETING"},
+		Target:     []string{"DELETED"},
+		Refresh:    deleteNicRefreshFunc(computeClient, serverID),
 		Timeout:    d.Timeout(schema.TimeoutDelete),
-		Delay:      5 * time.Second,
-		MinTimeout: 5 * time.Second,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Second,
```

Comment:
```suggestion
		PollInterval : 10 * time.Second,
```
The maximum interval is 10s and does not increase. It is equivalent to `PollInterval`. Therefore, `PollInterval` is recommended in this case.